src/iosrv/README.md

```diff
@@ -2,13 +2,33 @@
 
 ## Module Interface
 
-## Argobot Integration
+The I/O server supports a module interface that allows to load server-side code on demand. Each module is effectively a library dynamically loaded by the I/O server via dlopen.
+The interface between the module and the I/O server is defined in the `dss_module` data structure.
+
+Each module should specify:
+- a module name
+- a module identifier from `daos_module_id`
+- a feature bitmask
+- a module initialization and finalize function
+
+In addition, a module can optionally configure:
+- a setup and cleanup function invoked once the overall stack is up and running
+- CART RPC handlers
+- dRPC handlers
+
+## Thread Model & Argobot Integration
+
+The I/O server is a multi-threaded process using Argobots for non-blocking processing.
+
+By default, one main xstream plus two offload xstreams are created per target. Moreover, an extra xtream is created to handle incoming metadata requests. Each xstream is bound to a specific CPU core. The main xstream is the one receiving incoming target requests from both client and the other servers. A specific ULT is started to make progress on network and NVMe I/O opeations.
```

Comment:
Two offload streams by default, which is configurable by daos_server parameters.